src/mysql.c

```diff
@@ -238,6 +240,13 @@ static int mysql_config (oconfig_item_t *ci) /* {{{ */
 
 static MYSQL *getconnection (mysql_database_t *db)
 {
+	// This is necessary because we might be running on a different
+	// thread than before.
+	static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+	pthread_mutex_lock(&mutex);
+	mysql_library_init(0, NULL, NULL);
```

Comment:
Aha -- so the 2nd point at the bottom of http://dev.mysql.com/doc/refman/5.1/en/c-api-threaded-clients.html

So cleaner would be doing one of two things I think
* Call mysql_library_init in a plugin_init() and then call mysql_thread_init() in the read callback.  It's a smaller set of work potentially being done but still kind of ugly.  Better would be if there was a callback when threads are created but I don't actually see one.
* Or set up a plugin context to be used just by the mysql plugin so that reads don't occur across threads.  This is probably the right upstream thing (and is done by some of the other plugins)